### PR TITLE
SHARD-2392 - StringifyReduce unit tests

### PR DESCRIPTION
**Pull request by mhanson-github**
### **PR Type**
Tests

___

### **Description**
- Add comprehensive unit tests for `stringifyReduce` function

- Test `makeShortHash` function with various input scenarios

- Mock `safeStringify` from external dependency for isolation

- Cover edge cases: circular refs, deep/large objects, custom types

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>StringifyReduce.test.ts</strong><dd><code>Add full unit test suite for StringifyReduce utilities</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

test/unit/src/profiler/StringifyReduce.test.ts

<li>Introduces extensive unit tests for <code>stringifyReduce</code> and <code>makeShortHash</code><br> <li> Mocks external <code>safeStringify</code> for controlled test outputs<br> <li> Covers primitives, arrays, objects, Maps, Sets, Dates, RegExp, custom <br>classes, and edge cases<br> <li> Validates handling of undefined, functions, circular refs, and <br>deep/large objects

</details>

  </td>
  <td><a href="https://github.com/shardeum/archiver/pull/252/files#diff-0508516da2867728f8b463e4a283b1493cd3b2880a70c7895161479938cd9bf9">+406/-0</a>&nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

___

> <details> <summary>  Need help?</summary><li>Type <code>/help how to ...</code> in the comments thread for any questions about PR-Agent usage.</li><li>Check out the <a href="https://qodo-merge-docs.qodo.ai/usage-guide/">documentation</a> for more information.</li></details>

### CODE REVIEW

File: test/unit/src/profiler/StringifyReduce.test.ts [hunk 1]
```diff
@@ -0,0 +1,550 @@
...
+    })
+
+    it('should handle mixed character types correctly', () => {
+      const str = 'a🌟b'.repeat(16) // 64 chars
+      const result = makeShortHash(str)
+      expect(result.length).toBeLessThan(str.length)
+      expect(result).toContain('a')
+      expect(result).toContain('🌟')
+    })
+  })
```
Comment (aniketdivekar) [inline test/unit/src/profiler/StringifyReduce.test.ts:106]: Is our code supposed to handle emojis, or should it throw an error?

**Review by aniketdivekar — APPROVED** [review]

File: test/unit/src/profiler/StringifyReduce.test.ts (hunk 1 above)
Comment (mhanson-github) [reply test/unit/src/profiler/StringifyReduce.test.ts:106]: As the code is written it does nothing but shorten strings of specific lengths.  It should be rewritten but that's a larger scope task.
